src/multipass.ts

```diff
@@ -1,25 +1,54 @@
-import { ethers, Wallet } from "ethers";
-import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
-import { LibMultipass } from "@peeramid-labs/multipass/types/src/Multipass";
-import { RegisterMessage } from "./types";
-import { chainIdMapping, getArtifact, SupportedChains } from "./utils";
+import {
+  type Address,
+  type PublicClient,
+  type WalletClient,
+  type Hex,
+  getContract,
+  stringToHex,
+  zeroAddress,
+  isAddress,
+  SignTypedDataParameters,
+  TypedDataDomain,
+} from "viem";
+import { getArtifact, type SupportedChains } from "./utils";
+import multipassAbi from "@peeramid-labs/multipass/abi/src/Multipass.sol/Multipass";
+import type { RegisterMessage } from "./types";
+
+export type NameQuery = {
+  name: Hex;
+  id: Hex;
+  wallet: Address;
+  domainName: Hex;
+  targetDomain: Hex;
+};
+
 export default class Multipass {
   private chainId: string;
   private name: string;
   private version: string;
-  constructor({ chainName }: { chainName: SupportedChains }) {
-    this.chainId = chainIdMapping[chainName];
-    const c = getArtifact(chainName, "Multipass");
-    this.name = c.execute.args[0];
-    this.version = c.execute.args[1];
+  private publicClient: PublicClient;
+  private walletClient: WalletClient;
+  private instanceAddress: Address;
+
+  constructor({
+    chainName,
+    publicClient,
+    walletClient,
+  }: {
+    chainName: SupportedChains;
+    publicClient: PublicClient;
+    walletClient: WalletClient;
+  }) {
+    const artifact = getArtifact(chainName, "Multipass");
+    this.chainId = chainName;
```

Comment:
this wrong chainName: SupportedChains; can't be set to chainId. I guess we need to modify constructor of Multipass to 
    chainId: string
    publicClient: PublicClient;
    walletClient: WalletClient;